### PR TITLE
adding fastqc report variables

### PR DESCRIPTION
**Pull request by vicfabienne**
! New input argument ! snakemake change needed: The report has two links that are supposed to link to the fastqc report htmls. I added now the variables for that and the locations where (according to the snakefile) the fastqc output goes. @Jan I think that needs to be changed to output htmls. Furthermore it needs to run twice (before and after trimming). @Miri @Jan this rmd now gets a new argument: fastqc_dir. I only changed it in the "input settings" chunk since I think the "params" header changed anyways with Miriams-renderSite-version

### CODE REVIEW

File: scripts/qc_report_per_sample.rmd [hunk 1]
```diff
@@ -20,11 +20,18 @@ library(stringr)
...
 # Input Settings
 ```{r InputSettings, echo = FALSE, include = FALSE}
 coverage_file <- params$coverage_file
 sample_name <- params$sample_name
 outFile <- params$outFile
+fastqc_dir <- params$fastqc_dir # TODO @Jan @Miri add that argument to the snakemake rule and the rendersite function?
+# TODO: @Jan snakemake needs to be run twice before and after the preprocessing
+# TODO: @Jan @Ricardo 
+fastqc_unprocessed <- "FASTQ_DIR/{sample}_unprocessed_fastqc.html"
+fastqc_processed <- "FASTQ_DIR/{sample}_unprocessed_fastqc.html"
```
Comment (rekado) [inline scripts/qc_report_per_sample.rmd]: should be "_processed_fastqc.html", no?

Comment (rekado) [reply scripts/qc_report_per_sample.rmd]: FASTQ_DIR is a string, but I guess you want it to be a variable, such as the earlier `fastqc_dir`, no?

File: scripts/qc_report_per_sample.rmd [hunk 2]
```diff
@@ -20,11 +20,18 @@ library(stringr)
...
 library(openssl)
 ```
+
 # Input Settings
 ```{r InputSettings, echo = FALSE, include = FALSE}
 coverage_file <- params$coverage_file
 sample_name <- params$sample_name
 outFile <- params$outFile
+fastqc_dir <- params$fastqc_dir # TODO @Jan @Miri add that argument to the snakemake rule and the rendersite function?
+# TODO: @Jan snakemake needs to be run twice before and after the preprocessing
```
Comment (rekado) [inline scripts/qc_report_per_sample.rmd:30]: What are the inputs for each of the runs?

Comment (vicfabienne) [reply scripts/qc_report_per_sample.rmd:30]: What do you mean?

File: scripts/qc_report_per_sample.rmd (hunk 1 above)
Comment (vicfabienne) [reply scripts/qc_report_per_sample.rmd]: fixed

File: scripts/qc_report_per_sample.rmd [hunk 3]
```diff
@@ -20,11 +20,18 @@ library(stringr)
...
+# TODO: @Jan @Ricardo 
+fastqc_unprocessed <- paste0(FASTQ_DIR,"/{sample}_unprocessed_fastqc.html")
+fastqc_processed <- paste0(FASTQ_DIR,"/{sample}_unprocessed_fastqc.html")
```
Comment (rekado) [inline scripts/qc_report_per_sample.rmd:33]: Please use `file.path` instead of `paste0`.

File: scripts/qc_report_per_sample.rmd (hunk 2 above)
Comment (rekado) [reply scripts/qc_report_per_sample.rmd:30]: You wrote "snakemake needs to be run twice before and after the preprocessing".  How do these runs differ in terms of inputs and outputs?

File: scripts/variantreport_p_sample.rmd [hunk 4]
```diff
@@ -180,6 +180,16 @@ getIdentCols <- function (dataframe) {
...
+
+dedupeMuts <- function (mut, sigmut.df, dedup.df) {
+        # get variant group per mutation
+        duped_muts <- grep (mut, sigmut.df$value)
+        grouped <- sigmut.df$name[duped_muts]
+        groupName <- paste(grouped, collapse = ",")
+      
+        dedup.df$name[dedup.df$value == mut] <- groupName
+        return (dedup.df)
+}
```
Comment (rekado) [inline scripts/variantreport_p_sample.rmd:192]: Is this procedure a duplicate of what has already landed in `main`?

File: scripts/qc_report_per_sample.rmd (hunk 2 above)
Comment (vicfabienne) [reply scripts/qc_report_per_sample.rmd:30]: in the first run the inputs are the raw_data.fastq files. 

In the second run the inputs would be the prinseq-outputs, the files with the preprocessed reads (are they bam already?)

Comment (vicfabienne) [reply scripts/qc_report_per_sample.rmd:30]: As output in both cases I need the fastqc_report.html files

File: scripts/variantreport_p_sample.rmd (hunk 4 above)
Comment (vicfabienne) [reply scripts/variantreport_p_sample.rmd:192]: I'm not sure what you mean.

Comment (rekado) [reply scripts/variantreport_p_sample.rmd:192]: https://github.com/BIMSBbioinfo/pigx_sarscov2_ww/blob/main/scripts/report_scripts/variantreport_p_sample.Rmd#L149

Comment (vicfabienne) [reply scripts/variantreport_p_sample.rmd:192]: Ah, only see now that it's tracked as added here. 
This is probably a result of me not properly pulling and rebasing. 

So yes, it's a duplicate. But I will probably just resolve this when I rebase I guess.

**Pull request closed without merging**